back/app/services/survey_results_generator_service.rb

```diff
@@ -164,19 +164,24 @@ def build_select_response(answers, field, group_field)
       answers: answers,
       multilocs: get_multilocs(field, group_field)
     })
+
     attributes[:textResponses] = get_text_responses("#{field.key}_other") if field.other_option_text_field
-    attributes[:legend] = group_field.options.map(&:key) + [nil] if group_field.present?
+    attributes[:legend] = generate_answer_keys(group_field) if group_field.present?
 
     attributes
   end
 
   def get_multilocs(field, group_field)
-    multilocs = { answer: get_option_details(field) }
-    multilocs[:group] = get_option_details(group_field) if group_field
+    multilocs = { answer: get_option_multilocs(field) }
+    multilocs[:group] = get_option_multilocs(group_field) if group_field
     multilocs
   end
 
-  def get_option_details(field)
+  def get_option_multilocs(field)
+    if field.input_type == 'linear_scale'
```

Comment:
We already have a method dedicated to `linear_scale` in this class, `visit_linear_scale`. Is it possible to check/decide it there?

Reply:
Sometimes we are visiting a select field, but slicing by a linear scale. This way, this function can handle all cases

Reply:
With this line, `field_attributes[:multilocs][:answer] = answer_titles`, don't we set this field to the same value two times?

Reply:
Good point. I'll delete it

Reply:
It's not a blocker for this PR, just thinking about possible (maybe future) improvements.